transform_aes.c

```diff
@@ -22,15 +22,18 @@
 #if defined(N2N_HAVE_AES)
 
 #include "openssl/aes.h"
+#include "openssl/sha.h"
 #ifndef _MSC_VER
 /* Not included in Visual Studio 2008 */
 #include <strings.h> /* index() */
 #endif
 
 #define N2N_AES_NUM_SA                  32 /* space for SAa */
 
-#define N2N_AES_TRANSFORM_VERSION       1  /* version of the transform encoding */
-#define N2N_AES_IVEC_SIZE               32 /* Enough space for biggest AES ivec */
+#define N2N_AES_TRANSFORM_VERSION       2  /* version of the transform encoding */
+#define N2N_AES_IVEC_SIZE               16 /* Enough space for biggest AES ivec */
+#define N2N_AES_MSGAUTH_SIZE		16
+#define HASH_SIZE			64
```

Comment:
64 is because of SHA512, please add a comment

Reply:
Okay, I will add a comment. Alternatively, I could just use SHA512_DIGEST_LENGTH throughout the code without definition here - found out about it later.

Reply:
Used SHA512_DIGEST_LENGTH.